From 17ce51ada744847356b38e5e179d38942d4499a2 Mon Sep 17 00:00:00 2001
From: Keith Packard <keithp@keithp.com>
Date: Sun, 6 Oct 2013 15:23:05 -0700
Subject: [PATCH 2/3] Ignore document errors during build

---
 gcc-4.8.3/gcc/genhooks.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gcc-4.8.3/gcc/genhooks.c b/gcc-4.8.1/gcc/genhooks.c
index 28bb226..179fc68 100644
--- a/gcc-4.8.3/gcc/genhooks.c
+++ b/gcc-4.8.3/gcc/genhooks.c
@@ -164,8 +164,8 @@ emit_documentation (const char *in_fname)
 	    fatal ("Duplicate hook %s\n", sh.name);
 	  shp->pos = i;
 	}
-      else
-	fatal ("No place specified to document hook %s\n", sh.name);
+//      else
+//	fatal ("No place specified to document hook %s\n", sh.name);
       free (sh.name);
     }
   /* Copy input file to stdout, substituting @hook directives with the
-- 
1.8.4.rc3

