diff --git a/src/gcc/config/arm/t-rmprofile b/src/gcc/config/arm/t-rmprofile
index 3939461..fe2e59b 100644
--- a/src/gcc/config/arm/t-rmprofile
+++ b/src/gcc/config/arm/t-rmprofile
@@ -7,20 +7,22 @@ space :=
 space +=
 
 MULTILIB_OPTIONS   = mthumb/marm
 MULTILIB_DIRNAMES  = thumb arm
 MULTILIB_OPTIONS  += march=armv6s-m/march=armv7-m/march=armv7e-m/march=armv7/mcpu=cortex-m7
 MULTILIB_DIRNAMES += armv6-m armv7-m armv7e-m armv7-ar cortex-m7
 MULTILIB_OPTIONS  += mfloat-abi=softfp/mfloat-abi=hard
 MULTILIB_DIRNAMES += softfp fpu
 MULTILIB_OPTIONS  += mfpu=fpv4-sp-d16/mfpu=vfpv3-d16/mfpu=fpv5-sp-d16/mfpu=fpv5-d16
 MULTILIB_DIRNAMES += fpv4-sp-d16 vfpv3-d16 fpv5-sp-d16 fpv5-d16
+MULTILIB_OPTIONS  += mbig-endian
+MULTILIB_DIRNAMES += be
 
 MULTILIB_MATCHES   = march?armv6s-m=mcpu?cortex-m0
 MULTILIB_MATCHES  += march?armv6s-m=mcpu?cortex-m0plus
 MULTILIB_MATCHES  += march?armv6s-m=mcpu?cortex-m1
 MULTILIB_MATCHES  += march?armv6s-m=march?armv6-m
 MULTILIB_MATCHES  += march?armv7-m=mcpu?cortex-m3
 MULTILIB_MATCHES  += march?armv7e-m=mcpu?cortex-m4
 MULTILIB_MATCHES  += march?armv7=march?armv7-r
 MULTILIB_MATCHES  += march?armv7=march?armv7-a
 MULTILIB_MATCHES  += march?armv7=mcpu?cortex-r4
@@ -83,17 +85,20 @@ MULTILIB_OSDIRNAMES += mthumb/mcpu.cortex-m7=!cortex-m7
 MULTILIB_OSDIRNAMES += mthumb/mcpu.cortex-m7/mfloat-abi.hard/mfpu.fpv5-sp-d16=!cortex-m7/fpu/fpv5-sp-d16
 MULTILIB_OSDIRNAMES += mthumb/mcpu.cortex-m7/mfloat-abi.softfp/mfpu.fpv5-sp-d16=!cortex-m7/softfp/fpv5-sp-d16
 MULTILIB_OSDIRNAMES += mthumb/mcpu.cortex-m7/mfloat-abi.hard/mfpu.fpv5-d16=!cortex-m7/fpu/fpv5-d16
 MULTILIB_OSDIRNAMES += mthumb/mcpu.cortex-m7/mfloat-abi.softfp/mfpu.fpv5-d16=!cortex-m7/softfp/fpv5-d16
 endif
 
 ifneq (,$(filter armv7 armv7-r armv7-a,$(subst $(comma),$(space),$(with_multilib_list))))
 MULTILIB_REQUIRED   += mthumb/march=armv7
 MULTILIB_REQUIRED   += mthumb/march=armv7/mfloat-abi=softfp/mfpu=vfpv3-d16
 MULTILIB_REQUIRED   += mthumb/march=armv7/mfloat-abi=hard/mfpu=vfpv3-d16
+MULTILIB_REQUIRED   += mthumb/march=armv7/mfloat-abi=hard/mfpu=vfpv3-d16/mbig-endian
 MULTILIB_OSDIRNAMES += mthumb/march.armv7=!armv7-ar/thumb
 MULTILIB_OSDIRNAMES += mthumb/march.armv7/mfloat-abi.hard/mfpu.vfpv3-d16=!armv7-ar/thumb/fpu
 MULTILIB_OSDIRNAMES += mthumb/march.armv7/mfloat-abi.softfp/mfpu.vfpv3-d16=!armv7-ar/thumb/softfp
+MULTILIB_OSDIRNAMES += mthumb/march.armv7/mfloat-abi.hard/mfpu.vfpv3-d16=!armv7-ar/thumb/fpu/be
 MULTILIB_REUSE      += mthumb/march.armv7=marm/march.armv7
 MULTILIB_REUSE      += mthumb/march.armv7/mfloat-abi.softfp/mfpu.vfpv3-d16=marm/march.armv7/mfloat-abi.softfp/mfpu.vfpv3-d16
 MULTILIB_REUSE      += mthumb/march.armv7/mfloat-abi.hard/mfpu.vfpv3-d16=marm/march.armv7/mfloat-abi.hard/mfpu.vfpv3-d16
+MULTILIB_REUSE      += mthumb/march.armv7/mfloat-abi.hard/mfpu.vfpv3-d16=marm/march.armv7/mfloat-abi.hard/mfpu.vfpv3-d16/mbig-endian
 endif
